### PR TITLE
Add Autoscaling Spec Reference

### PR DESCRIPTION
**Pull request by jonathanmeier5**
*Issue #, if available:*
Epic: https://github.com/aws/eks-anywhere/issues/3186, https://github.com/aws/eks-anywhere/issues/3187
Ticket: https://github.com/aws/eks-anywhere/issues/3789

*Description of changes:*
Adds documentation for configuring autoscaling in your cluster specification.

*Testing (if applicable):*
Ran make serve-container and tested links, etc.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/reference/clusterspec/baremetal.md [hunk 1]
```diff
@@ -162,14 +162,20 @@ This takes in a list of node groups that you can define for your workers.
 You may define one or more worker node groups.
 
 ### workerNodeGroupConfigurations.count (required)
```
Comment (jiayiwang7) [inline docs/content/en/docs/reference/clusterspec/baremetal.md:164]:
```suggestion
### workerNodeGroupConfigurations.count
```

File: docs/content/en/docs/reference/clusterspec/cloudstack.md [hunk 2]
```diff
@@ -247,14 +247,20 @@ This takes in a list of node groups that you can define for your workers.
 You may define one or more worker node groups.
 
 ### workerNodeGroupConfigurations.count (required)
```
Comment (jiayiwang7) [inline docs/content/en/docs/reference/clusterspec/cloudstack.md:249]:
```suggestion
### workerNodeGroupConfigurations.count
```

File: docs/content/en/docs/reference/clusterspec/vsphere.md [hunk 3]
```diff
@@ -168,14 +168,20 @@ This takes in a list of node groups that you can define for your workers.
 You may define one or more worker node groups.
 
 ### workerNodeGroupConfigurations.count (required)
```
Comment (jiayiwang7) [inline docs/content/en/docs/reference/clusterspec/vsphere.md:170]:
```suggestion
### workerNodeGroupConfigurations.count
```

**Review by jiayiwang7 — APPROVED** [review]

File: docs/content/en/docs/tasks/cluster/cluster-scale/baremetal-scale.md [hunk 4]
```diff
@@ -47,3 +47,9 @@ To scale control plane nodes:
 ```bash
 kubectl scale kubeadmcontrolplane -n eksa-system <controlPlaneName> --replicas <num replicas>
 ```
+
+### Autoscaling
+
+EKS-A supports autoscaling of worker node groups using the [Kubernetes Cluster Autoscaler](https://github.com/kubernetes/autoscaler/) and as a [curated package](../../../../reference/packagespec/cluster-autoscaler/).
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/cluster/cluster-scale/baremetal-scale.md:53]:
```suggestion
EKS Anywhere supports autoscaling of worker node groups using the [Kubernetes Cluster Autoscaler](https://github.com/kubernetes/autoscaler/) and as a [curated package](../../../../reference/packagespec/cluster-autoscaler/).
```

File: docs/content/en/docs/tasks/cluster/cluster-scale/cloudstack-scale.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+linkTitle: "Scale CloudStack cluster"
+weight: 20
+date: 2017-01-05
+description: >
+  How to scale your CloudStack cluster
+---
+
+### Autoscaling
+
+EKS-A supports autoscaling of worker node groups using the [Kubernetes Cluster Autoscaler](https://github.com/kubernetes/autoscaler/) and as a [curated package](../../../../reference/packagespec/cluster-autoscaler/).
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/cluster/cluster-scale/cloudstack-scale.md:12]:
```suggestion
EKS Anywhere supports autoscaling of worker node groups using the [Kubernetes Cluster Autoscaler](https://github.com/kubernetes/autoscaler/) and as a [curated package](../../../../reference/packagespec/cluster-autoscaler/).
```

File: docs/content/en/docs/tasks/cluster/cluster-scale/vsphere-scale.md [hunk 6]
```diff
@@ -66,13 +66,8 @@ You can do this by storing your cluster config manifest in git and then having a
 Or you can use a GitOps controller to apply the changes.
 To read more about making changes with the integrated Flux GitOps controller you can read how to [Manage a cluster with GitOps]({{< relref "../cluster-flux" >}}).
 
-### Automatic scaling
+### Autoscaling
 
-Automatic cluster scaling is designed for worker nodes and it is not advised to automatically scale your control plane.
-Typically, autoscaling is done with a controller such as the [Kubernetes Cluster Autoscaler](https://github.com/kubernetes/autoscaler/).
-This has some concerns in an on-prem environment.
+EKS-A supports autoscaling of worker node groups using the [Kubernetes Cluster Autoscaler](https://github.com/kubernetes/autoscaler/) and as a [curated package](../../../../reference/packagespec/cluster-autoscaler/).
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/cluster/cluster-scale/vsphere-scale.md:71]:
```suggestion
EKS Anywhere supports autoscaling of worker node groups using the [Kubernetes Cluster Autoscaler](https://github.com/kubernetes/autoscaler/) and as a [curated package](../../../../reference/packagespec/cluster-autoscaler/).
```

**Review by chrisnegus — COMMENTED** [review]
A couple of small suggestions. Otherwise looks good.

**Pull request merged**